File: app/Models/Suggestion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Suggestion extends Model
{
    protected $table = 'suggestions';

    protected $fillable = [
        'usuario_id',
        'titulo',
        'descripcion',
        'estado',
        'conteo_votos',
    ];

    // Agregamos el atributo calculado para que se incluya en la respuesta JSON
    protected $appends = ['userVoted'];

    /**
     * Relación: Una sugerencia tiene muchos votos.
     */
    public function votes()
    {
        return $this->hasMany(SuggestionVote::class, 'suggestion_id');
    }

    /**
     * Accesor para determinar si el usuario autenticado ya ha votado esta sugerencia.
     * Se consulta si existe un voto en la relación "votes" filtrado por el ID del usuario.
     */
    public function getUserVotedAttribute()
    {
        $userId = auth()->user() ? auth()->user()->id_empleado : null;
        if ($userId) {
            return $this->votes()->where('usuario_id', $userId)->exists();
        }
        return false;
    }

    /**
     * Relación: La sugerencia pertenece a un usuario (el autor).
     */
    public function user()
    {
        return $this->belongsTo(User::class, 'usuario_id', 'id_empleado');
    }
}

File: android/app/src/main/java/es/bomberosgranada/app/data/models/Suggestion.kt
package es.bomberosgranada.app.data.models

data class Suggestion(
    val id: Int,
    val usuario_id: Int,
    val titulo: String,
    val descripcion: String? = null,
    val estado: String? = null,
    val conteo_votos: Int,
    val fecha_creacion: String,
    val fecha_actualizacion: String? = null,
    val userVoted: Boolean? = null,
    // Relación
    val user: User? = null
)

data class CreateSuggestionRequest(
    val usuario_id: Int,
    val titulo: String,
    val descripcion: String? = null,
    val estado: String? = "pendiente",
    val conteo_votos: Int? = 0
)

data class UpdateSuggestionRequest(
    val usuario_id: Int,
    val titulo: String,
    val descripcion: String? = null,
    val estado: String? = null,
    val conteo_votos: Int? = null
)

data class VoteSuggestionRequest(
    val suggestion_id: Int,
    val usuario_id: Int
)

data class VoteSuggestionResponse(
    val message: String,
    val conteo_votos: Int? = null,
    val vote: SuggestionVote? = null
)

data class SuggestionVote(
    val id: Int,
    val suggestion_id: Int,
    val usuario_id: Int,
    val created_at: String,
    val updated_at: String? = null
)